src/p2p/p2p.cpp

```diff
@@ -49,9 +56,24 @@ void start_peer_connections()
 {
     auto address = net::ip::make_address(conf::cfg.listenip);
 
+    // Adding ssl context options disallowing requests which supports sslv2 and sslv3 which have security vulnerabilitis
```

Comment:
Can't we move this entire section configuring sslcontext to inside the socket_server? The same code is now duplicated in user listener as well.